terraform/eks-cloudwatch/main.tf

```diff
@@ -0,0 +1,158 @@
+# ------------------------------------------------------------------------
+# Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License").
+# You may not use this file except in compliance with the License.
+# A copy of the License is located at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# or in the "license" file accompanying this file. This file is distributed
+# on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+# express or implied. See the License for the specific language governing
+# permissions and limitations under the License.
+# -------------------------------------------------------------------------
+
+# creating a eks cluster takes around 10 minutes typically.
+# so in the eks/k8s test, we need tester to provide the cluster instead of creating it in terraform
+# so that we can shorten the execution time
+
+variable "aws_account_id" {
+  default = ""
+}
+
+variable "sample_app_image_repo" {
+  default = "611364707713.dkr.ecr.us-west-2.amazonaws.com/otel-test/container-insight-samples"
+}
+
+module "common" {
+  source = "../common"
+
+  aoc_image_repo = var.aoc_image_repo
+  aoc_version = var.aoc_version
+}
+
+# region
+provider "aws" {
+  region = var.region
+}
+
+# get eks cluster by name
+data "aws_eks_cluster" "testing_cluster" {
+  name = var.eks_cluster_name
+}
+
+data "aws_eks_cluster_auth" "testing_cluster" {
+  name = var.eks_cluster_name
+}
+
+# set up kubectl
+provider "kubernetes" {
+  config_path = "~/.kube/config"
+  version = "~> 1.13"
+}
+
+# create a unique namespace for each run
+resource "kubernetes_namespace" "aoc_ns" {
```

Comment:
would that be possible we can somehow extract the existing `eks` module a bit so that we can reuse some of the common things?

Reply:
https://github.com/aws-observability/aws-otel-test-framework/pull/247#discussion_r600981387

Let me know if you have other suggestions.